testdata.log

```diff
@@ -8412,23 +8412,17 @@ SELECT * FROM t WHERE  EXISTS ( SELECT * FROM t WHERE i == 2 ) ORDER BY i;
 
 ---- 1349
 SELECT * FROM t WHERE  EXISTS ( SELECT * FROM t WHERE i == 2 );
```

Comment:
Not precisely. First of all, there are two SELECT keywords used in the statements, the plan shows one them (not sure which one).

The inner SELECT iterates *all* rows until it finds i == 2. If no match is found done. If a match is found, *all* rows are iterated to return the result set. See test no. 1348, for example.

> On the query above I assume the inner select evaluated to false hence no need to iterate at all.

The plan is created before execution. I don't understand how the plan can assume the results of the execution. But again, the proper function is what is important. The rendering of the plan is secondary, at least for now. We can look into this detail any time later.